### DISCUSSION TITLE
Ideas for driving product UI behavior based on workflow executions

### DISCUSSION THREAD

**Discussion by vnmnm** [elsa-workflows/elsa-core#4481] [Design & tech]
Hi,
From the Elsa UI, a user can configure the workflow with a sequence of activities. Let's say that some activity within the workflow requires user intervention on the UI (this is different UI than the Elsa UI where the workflow was configured) like say a button appears on the product UI when the workflow instance reaches that activity and user needs to click that button for the workflow to continue further. For example, an Approval Activity which displays `Approve` and `Reject` buttons on the UI and based on what the user clicks the workflow captures that input and resumes further. Is such an interaction supported by any of the Elsa activities today that I can take a look at for some guidance?

**Comment by sfmskywalker** [comment]
This sounds a lot like what the UserTask activity is about, which is inspired on the UserTask activity from Orchard Core. The activity works like this:

- Add a UserTask activity to the workflow.
- Configure it with a set of possible actions for the user to take, e.g. "Approve" and "Reject".
- When the activity executes, it creates a bookmark that includes the possible actions.
- Your UI app queries the Elsa DB (either directly or through a REST API) for all bookmarks for a given workflow instance and filters out just the UserTask bookmarks and uses the possible actions to display e.g. buttons.

Elsa 2 has this implemented, but I'm not sure what the status is of this activity in Elsa 3.
